### ISSUE TITLE
Support for builds 22621 and 22622 (Windows 11 22H2)

### ISSUE THREAD

**Issue by valinet** [valinet/ExplorerPatcher#1082]
Things currently broken in EP on Insider Preview Dev builds:

* Taskbar being fully transparent (fixed for now)
* Taskbar thumbnails not working (fixed)
* Disable Snap quadrants not working
* Disable Recommended section in Windows 11 Start menu not working (the live visual tree of the menu probably changed again; it should be an easy fix in lvt.c once the new visual tree is inspected) (this is a phony report, just checked on it, it seems to work fine)
* When pressing volume keys, no volume flyout is shown, because the old one was probably tossed away and no one linked the new one to the old shell; maybe enabling the old one would still be cool, idk, as usual, it has to be investigated
* WiFi flyout does not work anymore, I think (the patch for it is probably broken, the pattern I was searching for may not be there in the binary anymore - InjectShellExperienceHost is the starting point and the article here: [Functional Windows 10 flyouts in Windows 11](https://valinet.ro/2021/11/18/Functional-Windows-10-flyouts-on-Windows-11.html))
* Toasts and parts of the new shell seem to display on top of the taskbar, yet another thing they broke
* Win-Tab seems not to work anymore
* Disable rounded corners is broken
* Windows 11 taskbar context menu provided by ExplorerPatcher is not working anymore

@Simplestas Can you help me with any of these? What insight do you have on what has changed that the taskbar is disregarded by the new flyouts (they show on top of it)? Also, any idea how to make the new volume pop up work? And how to make the Win-Tab shortcut and Task View button work again? Thank you.

**Comment by lordmilko** [comment]
On Windows 22H2 Dev Build 22581.100 I found that
* if Aero Snap quadrants disabled prior to installing that insider version they continued to work as expected
* Upon restoring Aero Snap quadrants, they did not restore after restarting Windows Explorer - rather, they only worked again after rebooting my computer
* Upon trying to re-disable Aero Snap quadrants however, this did not work even after restarting the computer. 

Continuing to investigate

Another ExplorerPatcher feature broken in this build: when I right click the Windows 11 style taskbar in this build all I get is "Taskbar settings" instead of the old style context menu with Task Manager/Properties/etc. I don't think this is covered in your list above

**Comment by valinet** [comment]
Good to know.

Yeah, the Windows 11 taskbar context menu is provided by ExplorerPatcher in the 22000-based builds. Yeah, something changed in the implementation, that has to be fixed as well, I will add it to the list above.

**Comment by lordmilko** [comment]
From what I can see it appears the reason that disabling snap quadrants no longer works is because the method `twinui.pcshell!SnapLayoutHelpers::InitializeSnapQuadrantsPolicy` no longer exists. However, the method `SnapLayoutHelpers::AreQuadrantsEnabled` *does* still exist, as does the global variable that stores the result of the snap quadrants policy. While I cannot see any other references to this variable (which raises the question of whether it's even still set anywhere) the implementation of `CShellSnapComponent::PerformDelayedInitialization` seems to have a number of differences, and I can't seem to get WinDbg working even with the latest preview SDK; in any case, if worst comes to worst a solution here could be to set this global variable ourselves or even hook `SnapLayoutHelpers::AreQuadrantsEnabled` directly.

IMO it will be best to leave this one until these changes to Aero Snap's implementation roll out to release so it's clear what options are available

**Comment by valinet** [comment]
> IMO it will be best to leave this one until these changes to Aero Snap's implementation roll out to release so it's clear what options are available

Yup, I agree 100%. And we use this list and post here to track these issues and hopefully we will solve most of them. Indeed, I hate if we duplicate work, fixing it this week might render it totally useless the next one.

> I can't seem to get WinDbg working even with the latest preview SDK

Mhm, happens on my end to, I don’t really understand what is the problem. “Access denied” when doing “GetContextState” or something like that.

**Comment by valinet** [comment]
### Status on builds 22621 and 22622

Info about the program's features on these builds and the roadmap for fixing whatever's broken.

✔️ Volume keys show new volume slider
❓ Brightness keys: untested (VM)
❌ Some crashes on Win+E presses.
❌ Taskbar items disappear when refreshing the taskbar using current method. Happens for toggling search, task view, center/left aligned, always combine/never combine, large/small icons, weather widget. Needs fixing in code.

**Taskbar**

✔️ Taskbar style: Windows 10 
❌ Defaults in Windows 11 22621 and 22622: Search button - off, Task view - on, Cortana - off. Needs fixing in the GUI.
❌ Cortana button doesn't work anymore, should be removed from the GUI.
❓ Start button style works only after a restart.
✔️ Taskbar alignment
✔️ Never combine/Always combine
✔️ Taskbar size: Large and Small

**System tray**

✔️ Skin taskbar and tray pop-up menus
✔️ Center tray icon pop-up menus
✔️ Flyout tray icon popups
✔️ Show touch keyboard
✔️ Show seconds in the clock (also refreshes correctly the taskbar, could be used to refresh it this way every other place it is broken)
✔️ Control center hide/show
✔️ Show desktop, all settings
❓ Open network and Internet settings from popup menu: all settings work, but CPL options sometimes crash Explorer.
✔️ Network: Control Center
✔️ Network: Windows 11 flyout
✔️ Network: Windows 10 flyout
✔️ Network: Windows 8 flyout (but half broken by design)
✔️ Network: Settings app
❓ Network: CPL options sometimes crash Explorer
✔️ Sound: Windows 10 flyout
✔️ Sound: Windows 7 flyout
✔️ Clock: Windows 10 flyout
✔️ Clock: Windows 11 flyout
✔️ Clock: Windows 7 flyout
❓ Battery: untested (VM)
❓ Language switcher: all Windows 10 options work; Windows 11 option weirdly enough produces the Windows 10 flyout from Windows 10 19044 with working animation instead of the Windows 11 flyout. Win+Space produces the Windows 10 side flyout.

**File Explorer**

❓ Register as shell extension: untested (deprecated)
✔️ Disable Windows 11 context menu
✔️ Always use legacy file transfer dialog
✔️ Use classical drive groupings in This PC
❌ Control interface: only Windows 11 command bar works. Won't fix, deprecated.
✔️ Immersive menus on/off
✔️ Disable navigation bar
✔️ Disable modern search bar
❌ Hide search bar completely
❌ Title bar: will be removed, since the title bar now contains tabs
❓ Mica effect: maybe reword options a bit; title bar and navigation bar looks weird. Also, windows glitch out after a while. Should be removed.

**Start menu**
Windows 11
✔️ Position on screen
✔️ Disable Recommended
❓ Open on monitor: untested (VM)
✔️ All apps default
❓ Maximum frequent apps to show: untested (VM)

Windows 10
✔️ Position on screen
✔️ Show more tiles
✔️ Corner preference
✔️ Display mode
✔️ App list
✔️ Pin tiles to Windows 10 Start menu

**Window switcher**
✔️ Windows 11 style
❌ Windows 10 style (maybe it could be revived, I know that they removed the "IsUndockingAsset" available feature which I used to hook in order to change the style, but maybe another hook/strategy is possible; not a big deal since sws exists anyway)
✔️ Windows NT
✔️ sws

**Weather**
✔️ Works

**Other**
✔️ Remember last used section
✔️ Win+C clock
✔️ Show separators between taskbar toolbars
❌ Show cmd instead of PowerShell in Win+X. Actually, it never worked properly in Windows 11. Will be removed and left as an option only on Windows 10.
✔️ Add shortcut to program settings in Win+X
✔️ Remove shortcut key from program settings item in Win+X
✔️ Disable Office hotkeys
✔️ Disable Win+F hotkey
❌ Disable rounded corners. Ugh, here we go again, probably needs a different patching strategy.
✔️ Disable quadrants when snapping windows
❌ Snap Assist: Windows 10 style not working, same reason as for Windows 10 window switcher
✔️ Alt+F4 default action
✔️ Redirects work

**Updates**
✔️ Works

**Advanced**
✔️ Pinned items act as quick launch
✔️ Remove extra gap when taskbar shows labels
❓Everything else is untested

**Comment by valinet** [comment]
I will close the issue since I reached a usable pre-release version.
